### DISCUSSION TITLE
More than 8 ports per node

### DISCUSSION THREAD

**Discussion by jaimeatstariongroup** [Blazor-Diagrams/Blazor.Diagrams#278] [General]
Is there a nice way to show more than 8 ports per node?

**Comment by zHaytam** [comment]
You have complete control on how to show them, after all you controll their display.

**Reply by jaimeatstariongroup** [reply]
Do you have an example? Because now I'm trying to display the ports in a circle around the node.

`       var angle = Math.PI / 2.0;
        var angleIncrement = (2.0 * Math.PI) / this.Node.Ports.Cast<DiagramPort>().Count();

        foreach (var port in this.Node.Ports.Cast<DiagramPort>())
        {
            var r = 200.0;
            var x = this.Node.Position.X - r * Math.Cos(angle);
            var y = this.Node.Position.Y - r * Math.Sin(angle);

            var posx = x.ToString() + "px";
            var posy = y.ToString() + "px";

            <PortRenderer Port="port"  Class="@(port.HasComments ? "port-comment custom-port" : "custom-port")"></PortRenderer>
            
            angle += angleIncrement;
        }`

However I can not style the port renderer to give the item somethig like: _style="top:posy; left:posx;"_
If you have any ideas of how to achieve this that would be great.

**Reply by jaimeatstariongroup** [reply]
Nevermind... I wrapper the _PortRenderer_ in another _div_ and I'm styling that _div_. Thank you for the help!
